classes/table/csp_report.php

```diff
@@ -40,6 +40,30 @@
  * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 class csp_report extends \table_sql {
+
+    /**
+     * Embeds a link to a drilldown table showing only 1 violation class
+     *
+     * @param stdObject $record fieldset object of db table with field timecreated
+     * @return string Link to drilldown table
+     */
+    protected function col_failcounter($record) {
+        // Get blocked URI, and set as param for page if clicked on
+        $url = new \moodle_url('/local/csp/csp_report.php', array ('viewviolationclass' => $record->blockeduri, 'sesskey' => sesskey()));
```

Comment:
Is sesskey really needed here?

Generally you want that for anything which mutates state, this prevents it being bookmarked or deep linked blindly.